src/lexer.l

```diff
@@ -208,13 +209,13 @@ struct|union|enum       {
   .                     { unput(yytext[0]); yy_pop_state(yyscanner); }
 }
 <STRUCT_AFTER_COLON>{
-  "{"|","|")"           {
-                          yy_pop_state(yyscanner);
-                          unput(yytext[0]);
-                          return Parser::make_IDENT(struct_type + " " + trim(buffer), loc);
-                        }
-  .                     { buffer += yytext[0]; }
-  \n                    buffer += '\n'; loc.lines(1); loc.step();
+  "{"|","|")"|"*"|";"|"[" {
```

Comment:
Should we invert this and instead of trying to think of all characters that aren't allow, just say only `ident` characters are allowed (with whitespace before the ident)?

Reply:
Good idea. Will update.